include/cantera/thermo/PengRobinson.h

```diff
@@ -0,0 +1,313 @@
+                         const std::string& species_j, double a);
+
+protected:
+    // Special functions inherited from MixtureFugacityTP
+    virtual double sresid() const;
+    virtual double hresid() const;
+
+    virtual double liquidVolEst(double TKelvin, double& pres) const;
+    virtual double densityCalc(double TKelvin, double pressure, int phase, double rhoguess);
+
+    virtual double densSpinodalLiquid() const;
+    virtual double densSpinodalGas() const;
+    virtual double dpdVCalc(double TKelvin, double molarVol, double& presCalc) const;
+
+    // Special functions not inherited from MixtureFugacityTP
+
+    //! Calculate temperature derivative d(a*alpha)/dT
+    /*!
+     *  These are stored internally.
+     */
+    double daAlpha_dT() const;
+
+    //! Calculate second derivative d2(a*alpha)/dT2
+    /*!
+     *  These are stored internally.
+     */
+    double d2aAlpha_dT2() const;
+
+public:
+
+    //! Calculate dpdV and dpdT at the current conditions
+    /*!
+     *  These are stored internally.
+     */
+    void calculatePressureDerivatives() const;
+
+    //! Update the a, b and alpha parameters
+    /*!
+     *  The a and the b parameters depend on the mole fraction and the
+     *  parameter alpha depends on the temperature. This function updates
+     *  the internal numbers based on the state of the object.
+     */
+    virtual void updateMixingExpressions();
+
+    //! Calculate the a, b and the alpha parameters given the temperature
+    /*!
+     * This function doesn't change the internal state of the object, so it is a
+     * const function.  It does use the stored mole fractions in the object.
+     *
+     * @param aCalc (output)  Returns the a value
+     * @param bCalc (output)  Returns the b value.
+     * @param aAlpha (output) Returns the (a*alpha) value.
+     */
+    void calculateAB(double& aCalc, double& bCalc, double& aAlpha) const;
+
+    void calcCriticalConditions(double& pc, double& tc, double& vc) const;
+
+    //! Prepare variables and call the function to solve the cubic equation of state
+    int solveCubic(double T, double pres, double a, double b, double aAlpha,
+                   double Vroot[3]) const;
+protected:
+    //! Value of b in the equation of state
+    /*!
+     *  m_b is a function of the mole fractions and species-specific b values.
+     */
+    double m_b;
+
+    //! Value of a and alpha in the equation of state
+    /*!
+     *  m_aAlpha_mix is a function of the temperature and the mole fractions. m_a depends only on the mole fractions.
+     */
+    double m_a;
+    double m_aAlpha_mix;
+
+    // Vectors required to store a_coeff, b_coeff, alpha, kappa and other values for every species. Length = m_kk
+    vector_fp m_b_coeffs;
+    vector_fp m_kappa;
+    mutable vector_fp m_dalphadT;
+    mutable vector_fp m_d2alphadT2;
+    vector_fp m_alpha;
+
+    //Matrices for Binary coefficients a_{i,j} and {a*alpha}_{i.j} are saved in an Array form. Length =  (m_kk, m_kk)
```

Comment:
```suggestion
    // Matrices for Binary coefficients a_{i,j} and {a*alpha}_{i.j} are saved in an Array form. Length =  (m_kk, m_kk)
```

I think this line is too long as well.

Reply:
Corrected.